pkg/scheduler/controller/shoot/scheduler_control.go

```diff
@@ -299,6 +300,11 @@ func filterCandidates(shoot *gardencorev1beta1.Shoot, seedList []*gardencorev1be
 			continue
 		}
 
+		if allocatableShoots, ok := seed.Status.Allocatable[gardencorev1beta1.ResourceShoots]; ok && int64(seedUsage[seed.Name]) >= allocatableShoots.Value() {
```

Comment:
Can you add a unit test case for this condition?

Reply:
Will try to do it, shouldn't be too hard although there are currently no unit tests for this package at all.

Reply:
Hm, there is https://github.com/gardener/gardener/blob/master/pkg/scheduler/controller/shoot/scheduler_test.go (which probably should be renamed to `scheduler_control_test.go`), or?

Reply:
Sorry, I was simply too quick confused, looking in a different package. I will add the tests.

Reply:
Done